src/components/BenefitPackageIndividualPanel.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import {
+  Grid, Typography, IconButton, Tooltip,
+} from '@material-ui/core';
+import {
+  FormattedMessage,
+  PublishedComponent,
+  TextInput,
+  NumberInput,
+  FormPanel,
+  formatMessage,
+} from '@openimis/fe-core';
+import { Person } from '@material-ui/icons';
+import { injectIntl } from 'react-intl';
+import { withTheme, withStyles } from '@material-ui/core/styles';
+import { EMPTY_STRING, RIGHT_INDIVIDUAL_UPDATE } from '../constants';
+
+const styles = (theme) => ({
+  tableTitle: theme.table.title,
+  item: theme.paper.item,
+  paper: theme.paper.paper,
+  fullHeight: {
+    height: '100%',
+  },
+});
+
+function renderHeadPanelSubtitle(rights, intl, history, modulesManager, classes, individualUuid) {
+  const openIndividual = () => history.push(`/${modulesManager.getRef('individual.route.individual')}`
+    + `/${individualUuid}`);
+
+  return (
+    <Grid item>
+      <Grid container align="center" justify="center" direction="column" className={classes.fullHeight}>
+        <Grid item>
+          <Typography>
+            <FormattedMessage
+              module="socialProtection"
+              id="socialProtection.benefitPackage.IndividualDetailPanel.title"
+            />
+            { !!individualUuid && (
+            <Tooltip title={formatMessage(intl, 'socialProtection', 'benefitPackage.IndividualDetailPanel.tooltip')}>
+              <IconButton onClick={openIndividual} disabled={!rights.includes(RIGHT_INDIVIDUAL_UPDATE)}>
+                <Person />
+              </IconButton>
+            </Tooltip>
+            )}
+          </Typography>
+        </Grid>
+      </Grid>
+    </Grid>
+  );
+}
+
+class BenefitPackageIndividualPanel extends FormPanel {
+  // eslint-disable-next-line class-methods-use-this
+  createAdditionalField(jsonExt) {
```

Comment:
Can you could consider using object destructuring in the mapping function to directly extract the property and value without indexing the array elements? Please let me know what do you think about this? 
```
function createAdditionalField(jsonExt) {
  if (!jsonExt) return [];

  const additionalFields = JSON.parse(jsonExt);

  return Object.entries(additionalFields).map(([property, value]) => {
    const field = { [property]: value };
    const fieldType = typeof value;
    return { field, fieldType };
  });
}
```

Reply:
That's a good point. It makes the whole function more readable. Thanks!